lms/djangoapps/course_wiki/tests/tests.py

```diff
@@ -196,3 +197,26 @@ def test_create_wiki_with_long_course_id(self):
 
         resp = self.client.get(course_wiki_page, follow=True, HTTP_REFERER=referer)
         self.assertEqual(resp.status_code, 200)
+
+    @patch.dict("django.conf.settings.FEATURES", {'ALLOW_WIKI_ROOT_ACCESS': True})
+    def test_consent_required(self):
+        """
+        Test that enterprise data sharing consent is required when enabled for the various courseware views.
+        """
+        # Public wikis can be accessed by non-enrolled users, and so direct access is not gated by the consent page
+        course = CourseFactory.create()
+        course.allow_public_wiki_access = False
+        course.save()
+
+        # However, for private wikis, enrolled users must pass through the consent gate
+        # (Unenrolled users are redirected to course/about)
+        course_id = unicode(course.id)
+        wiki_slug = course.wiki_slug
+        self.login(self.student, self.password)
+        self.enroll(course)
+
+        for (url, status_code) in (
+            (reverse('course_wiki', kwargs={'course_id': course_id}), 302),
+            ('/courses/{}/wiki/'.format(course_id), 200),
```

Comment:
Nope, because that URL doesn't have a name.  I can add one if you like, but I think it's [deliberately unnamed](https://github.com/edx/edx-platform/blob/master/lms/urls.py#L175-L179).